internal/controller/impersonatorconfig/impersonator_config_test.go

```diff
@@ -1201,6 +1221,133 @@ func TestImpersonatorConfigControllerSync(t *testing.T) {
 			})
 		})
 
+		when("using external TLS secrets", func() {
```

Comment:
Need to make sure that the generated CA secret for the generated TLS serving cert is never deleted

Reply:
done